python-large-file-upload/main.py

```diff
@@ -116,10 +112,35 @@ def main():
         sendRequest(card, "hub.set", opts.hub_config)
         logging.info(f"HUB config: {opts.hub_config}")
 
-    uploader = (notecardDataTransfer.BinaryDataUploaderLegacy(card, opts.route, printFcn=logging.debug, timeout=opts.timeout)
-                if opts.legacy
-                else notecardDataTransfer.BinaryDataUploader(card, opts.route, printFcn=logging.debug, timeout=opts.timeout))
-    
+
+    if opts.web_req_type.upper() == "PUT":
+        req = "web.put"
+    elif opts.web_req_type.upper() == "POST":
+        req = "web.post"
+    else:
+        if opts.debug:
+            logging.debug(f"Error: Invalid web request type: {opts.web_req_type}, options are 'PUT' or 'POST'")
+        else:
+            print(f"Error: Invalid web request type: {opts.web_req_type}, options are 'PUT' or 'POST'")
+        sys.exit(1)

```

Comment:
Throw an exception instead of using `print` and `sys.exit(1)`.

```
raise(Exception(f"Error: Invalid web request type: {opts.web_req_type}, options are 'PUT' or 'POST'"))
```